src/Discord.Net.Interactions/Info/Commands/ComponentCommandInfo.cs

```diff
@@ -13,6 +13,9 @@ namespace Discord.Interactions
     /// </summary>
     public class ComponentCommandInfo : CommandInfo<CommandParameterInfo>
     {
+        /// <inheritdoc/>
+        public override CommandTargetType CommandTargetType => CommandTargetType.ButtonInteraction | CommandTargetType.SelectMenuInteraction;
```

Comment:
This is actually my main choking point. I want to be able to view if a component interaction is dropdown or button. It would maybe be a better option to introduce a property called HasValues? This would imply that the component is select menu, as there are values for this. 

The alternative could maybe be to make a property called Values which will be null if not applicable, Array<T> if set. This gives users the ability to easily grab the select menu values before the handler is entered?